### ISSUE TITLE
JetBrains: Load file content in parallel

### ISSUE THREAD

**Issue by vdavid** [sourcegraph/sourcegraph#37371]
Currently, we use [`fetch` on the main thread](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/jetbrains/webview/src/search/lib/blob.ts?L25&subtree=true) to load files. When checking the "Network" tab of DevTools, it looks like a staircase:

<img width="951" alt="CleanShot 2022-06-17 at 10 54 34@2x" src="https://user-images.githubusercontent.com/2552265/174264198-9254b361-e5fe-4ade-acf5-1ae89ed07627.png">

so this sort of fetching seems to be blocking. It results in sub-optimal UX: [40-sec demo](https://www.loom.com/share/2bdd66b8bb54458892d549f97bbc4e2a)

A potential solution is to use [web workers](https://www.loginradius.com/blog/engineering/adding-multi-threading-to-javascript-using-web-workers/) for the fetching.

**Comment by sourcegraph-bot-2** [comment]
Heads up @muratsu @jjinnii @ryankscott - the "team/integrations" label was applied to this issue.

**Comment by vdavid** [comment]
Using [AbortController](https://dev.to/bil/using-abortcontroller-with-react-hooks-and-typescript-to-cancel-window-fetch-requests-1md4) is another option to make the UX smoother. Thanks, @taras-yemets, for the idea!

**Comment by philipp-spiess** [comment]
Fetch is already non-blocking. I don't understand how moving this to a web worker would fix the issue. Are we _awaiting_ for the fetch at some place before we make the new request? My alternative hypothesis is that this is limited by the network (either a limit on how many requests can be made concurrently or an issue in the implementation of JCEF?).

**Comment by vdavid** [comment]
Yes, we are _await_ing for the fetch [right here](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/jetbrains/webview/src/search/lib/blob.ts?L25&subtree=true). But that shouldn't be blocking either, I mean, while the fetch is running, the JS engine is free to do whatever we want, this `await` is basically just adding a callback.

**Comment by philipp-spiess** [comment]
I don't think there's an issue in the JavaScript side of things unfortunately. In Standalone mode, parallel fetching already works:

![Image](https://user-images.githubusercontent.com/458591/177978769-e4a5687a-dbc9-4cbc-9d40-e4ccc408d3ee.png)

So this is likely a bottleneck in the JCEF bridge (perhaps the use of only one I/O thread is the issue here? idk).

I'll see if we can implement aborting instead to improve the UX!
